contracts/roles/AgentManager.sol

```diff
@@ -51,7 +53,7 @@ contract AgentManager is AgentRoles {
         IIdentity _onchainID
     ) external {
         require(
-            isTransferManager(address(_onchainID)) && _onchainID.keyHasPurpose(keccak256(abi.encode(msg.sender)), 1),
+            isTransferManager(address(_onchainID)) && _onchainID.keyHasPurpose(keccak256(abi.encode(msg.sender)), 2),
```

Comment:
Is this modification expected? I guess yes, because 2 is `EXECUTION` key, but just to be sure.